src/client/Client.cc

```diff
@@ -5799,6 +5817,7 @@ void Client::unmount()
 
   while (!mds_requests.empty()) {
     ldout(cct, 10) << "waiting on " << mds_requests.size() << " requests" << dendl;
+    flush_mdlog_sync();
```

Comment:
My point is reducing the messages/requests that trigger mdlog->flush to as few as possible